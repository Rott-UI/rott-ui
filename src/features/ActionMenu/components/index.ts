export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
