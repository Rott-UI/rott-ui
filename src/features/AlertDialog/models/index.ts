export * from './alertDialogButtonProps'
export * from './alertDialogContextModel'
export * from './alertDialogModel'
