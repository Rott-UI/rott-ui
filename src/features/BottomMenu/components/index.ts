export * from './BottomMenu'
export * from './BottomMenuItem'
