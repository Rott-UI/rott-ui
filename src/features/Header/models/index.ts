export * from './HeaderIconProps'
export * from './HeaderProps'
