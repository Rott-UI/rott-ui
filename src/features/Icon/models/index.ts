export * from './iconProps'
export * from './iconType'
export * from './iconMode'
