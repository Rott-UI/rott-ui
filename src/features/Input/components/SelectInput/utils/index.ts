export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
