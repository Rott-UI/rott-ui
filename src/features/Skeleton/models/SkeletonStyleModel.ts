export interface SkeletonStyleProps {
  width: number
  height: number
  radius?: number
}
