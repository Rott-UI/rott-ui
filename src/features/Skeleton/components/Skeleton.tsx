// React Imports
import React, {FC, useEffect, useMemo, useRef} from 'react'

// React Native Imports
import {Animated, Easing, StyleSheet} from 'react-native'

// Component Imports
import {Item} from '../../../components'

// Constant Imports
import {COLOURS} from '../../../constants'

// Util and Lib Imports
import {display} from '../../../utils'

// Model Imports
import {SkeletonStyleProps} from '../models'

// Package Imports
import LinearGradient from 'react-native-linear-gradient'

interface SkeletonProps extends SkeletonStyleProps {
  show: boolean
  noAnimation?: boolean
  testID?: string
}

/**
 *
 * @param width Genislik - **Zorunlu**
 * @param height Yukseklik - **Zorunlu**
 * @param radius Radius - `default: 4px`
 * @param show Renderlanma durumu - `default: false`
 * @returns
 */
export const Skeleton: FC<SkeletonProps> = ({
  testID,
  width,
  height,
  radius = 4,
  noAnimation = false,
  show = false,
}) => {
  const animatedValueRef = useRef(new Animated.Value(0)).current

  useEffect(() => {
    if (!show) return

    if (!noAnimation) {
      const loop = Animated.loop(
        Animated.timing(animatedValueRef, {
          toValue: 1,
          duration: 1400,
          easing: Easing.ease,
          useNativeDriver: false,
        })
      )
      loop.start()

      return () => loop.stop()
    }
  }, [show])

  const animatedGradientStyle = useMemo(() => {
    const animationWidth = display.px(width as number)

    return {
      ...StyleSheet.absoluteFillObject,
      flexDirection: 'row' as const,
      transform: [
        {
          translateX: animatedValueRef.interpolate({
            inputRange: [0, 1],
            outputRange: [-animationWidth, animationWidth],
          }),
        },
      ],
    }
  }, [])

  const getGradientProps = (_width: number) => ({
    start: {x: 0, y: 0},
    end: {x: 1, y: 0},
    style: {...StyleSheet.absoluteFillObject, _width},
  })

  if (!show) return null

  return (
    <Item
      testID={testID}
      width={width}
      height={height}
      borderRadius={radius}
      overflowHidden
      backgroundColor={COLOURS.GREY100}>
      <Animated.View style={animatedGradientStyle}>
        <LinearGradient
          {...getGradientProps(width!)}
          style={{
            width: width ? display.px(width) : undefined,
            height: height ? display.px(height) : undefined,
          }}
          colors={[COLOURS.GREY100, COLOURS.WHITE, COLOURS.GREY100]}
        />
      </Animated.View>
    </Item>
  )
}
