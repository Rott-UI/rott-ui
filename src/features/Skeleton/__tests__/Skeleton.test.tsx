// Component Imports
import {Skeleton} from '../components/Skeleton'
import {Item} from '../../../components/Item'
import {Label} from '../../../components/Label'

// Util and Lib Imports
import {render} from '../../../utils'

// Model Imports
import {SkeletonStyleProps} from '../models'

const testId = {
  itemTestId: 'item-test-id',
  skeletonTestId: 'skeleton-test-id',
}

const skeletonStyle: SkeletonStyleProps = {
  width: 80,
  height: 16,
}

const DummyText = 'Test Content Text'

describe('Skeleton -> Custom Component', () => {
  it('skeleton snapshotı ile eşleşmeli', () => {
    const {skeletonTestId} = testId
    const {width, height} = skeletonStyle
    const rendered = render(
      <Skeleton testID={skeletonTestId} show={true} width={width} height={height} />
    )
    expect(rendered).toMatchSnapshot()
  })

  it('item icerisinde cagirilan skeleton icin snapshot eslesmeli', () => {
    const {itemTestId, skeletonTestId} = testId
    const rendered = render(
      <Item
        testID={itemTestId}
        skeletonTestID={skeletonTestId}
        skeletonStyle={skeletonStyle}
        skeletonShow>
        <Label>{DummyText}</Label>
      </Item>
    )
    expect(rendered).toMatchSnapshot()
  })

  it('skeleton skeletonShow true ise skeleton ekranda olmali', () => {
    const {itemTestId, skeletonTestId} = testId
    const {getByTestId} = render(
      <Item
        testID={itemTestId}
        skeletonTestID={skeletonTestId}
        skeletonStyle={skeletonStyle}
        skeletonShow>
        <Label>{DummyText}</Label>
      </Item>
    )

    const skeleton = getByTestId(skeletonTestId)
    expect(skeleton).toBeOnTheScreen()
  })

  it('skeleton skeletonShow true ise content ekranda olmamali', () => {
    const {itemTestId, skeletonTestId} = testId
    const {queryByTestId} = render(
      <Item
        testID={itemTestId}
        skeletonTestID={skeletonTestId}
        skeletonStyle={skeletonStyle}
        skeletonShow>
        <Label>{DummyText}</Label>
      </Item>
    )

    const content = queryByTestId(DummyText)
    expect(content).not.toBeOnTheScreen()
  })

  it('skeleton skeletonShow false ise skeleton ekranda olmamali', () => {
    const {itemTestId, skeletonTestId} = testId
    const {queryByTestId} = render(
      <Item
        testID={itemTestId}
        skeletonTestID={skeletonTestId}
        skeletonStyle={skeletonStyle}
        skeletonShow={false}>
        <Label>{DummyText}</Label>
      </Item>
    )

    const skeleton = queryByTestId(skeletonTestId)
    expect(skeleton).not.toBeOnTheScreen()
  })

  it('skeleton skeletonShow false ise content ekranda olmali', () => {
    const {itemTestId, skeletonTestId} = testId
    const {getByText} = render(
      <Item
        testID={itemTestId}
        skeletonTestID={skeletonTestId}
        skeletonStyle={skeletonStyle}
        skeletonShow={false}>
        <Label>{DummyText}</Label>
      </Item>
    )

    const content = getByText(DummyText)
    expect(content).toBeOnTheScreen()
  })
})
