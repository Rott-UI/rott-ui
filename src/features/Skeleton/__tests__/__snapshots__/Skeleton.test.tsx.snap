// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton -> Custom Component item icerisinde cagirilan skeleton icin snapshot eslesmeli 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
  testID="item-test-id"
>
  <View
    backgroundColor="rgba(234, 239, 240, 1)"
    borderRadius={4}
    height={16}
    overflowHidden={true}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": "rgba(234, 239, 240, 1)",
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": 31,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": 31,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": "hidden",
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": 154,
        "zIndex": undefined,
      }
    }
    testID="skeleton-test-id"
    width={80}
  >
    <View
      collapsable={false}
      style={
        {
          "bottom": 0,
          "flexDirection": "row",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateX": -154,
            },
          ],
        }
      }
    >
      <BVLinearGradient
        colors={
          [
            4293586928,
            4294967295,
            4293586928,
          ]
        }
        endPoint={
          {
            "x": 1,
            "y": 0,
          }
        }
        locations={null}
        startPoint={
          {
            "x": 0,
            "y": 0,
          }
        }
        style={
          {
            "height": 31,
            "width": 154,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Skeleton -> Custom Component skeleton snapshotı ile eşleşmeli 1`] = `
<View
  backgroundColor="rgba(234, 239, 240, 1)"
  borderRadius={4}
  height={16}
  overflowHidden={true}
  style={
    {
      "alignItems": undefined,
      "backgroundColor": "rgba(234, 239, 240, 1)",
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": 31,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": 31,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": "hidden",
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": 154,
      "zIndex": undefined,
    }
  }
  testID="skeleton-test-id"
  width={80}
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -154,
          },
        ],
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          4293586928,
          4294967295,
          4293586928,
        ]
      }
      endPoint={
        {
          "x": 1,
          "y": 0,
        }
      }
      locations={null}
      startPoint={
        {
          "x": 0,
          "y": 0,
        }
      }
      style={
        {
          "height": 31,
          "width": 154,
        }
      }
    />
  </View>
</View>
`;
